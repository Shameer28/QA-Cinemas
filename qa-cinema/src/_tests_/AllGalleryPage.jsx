import AllGalleryPage from "../components/AllGalleryPage/AllGalleryPage";

describe("All Gallery Page Testing", () => {

    it("Snapshot Check" , () => {
        const isntance = create(<AllGalleryPage/>).toJSON();
        expect(instance).toMatchSnapshot();
    })

})