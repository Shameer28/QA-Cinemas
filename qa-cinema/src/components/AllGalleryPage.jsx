
import ListMovies from "./props/ListMovies"

const AllGalleryPage = () => {
    return (<div>
        <h1>All Movies Ever Of All Time</h1>
        <ListMovies url="http://localhost:3000/movies/getAll"/>
    </div>);
}
export default AllGalleryPage;