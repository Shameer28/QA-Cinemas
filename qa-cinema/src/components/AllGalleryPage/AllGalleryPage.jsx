
import { Container, Row } from "react-bootstrap";

import ListMovies from "./../sharedcomponents/ListMovies"


const AllGalleryPage = () => {

    return (
        <div className="background">
            <Container>
                <br></br>
                <h2 style={{ color: "white", fontWeight: "400", fontfamily: "isonormregular, sans-serif", letterSpacing: ".15em", textTransform: "uppercase", lineHeight: "1.1" }}>Now Showing</h2>
                <br></br>
                <div>
                    <Row>
                        <ListMovies url="http://localhost:3000/movies/getAll" />
                    </Row>
                </div>

                <br></br>

            </Container>
        </div>
    );

}

export default AllGalleryPage;