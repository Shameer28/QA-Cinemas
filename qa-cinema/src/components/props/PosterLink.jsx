const PosterLink = ({image, _id}) => {
    return ( <a href={"/movieDetail/"+ _id}> <img src={image} alt="Click for More Details"/></a> );
}
 
export default PosterLink;