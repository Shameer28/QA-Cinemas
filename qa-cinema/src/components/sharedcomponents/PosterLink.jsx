const PosterLink = ({image, _id, title}) => {
    return (
        <div>
        <a href={"/movieDetail/" + _id}> <img class="movie" src={image} alt="Click for More Details" /></a><br></br>
            <div style={{ color: "white", fontWeight: "400", fontfamily: "isonormregular, sans-serif", letterSpacing: ".15em", textTransform: "uppercase", lineHeight: "1.1", textAlign: "center", width: "75%" }}><br></br>{title}</div>
            </div>
    );
}
 
export default PosterLink;