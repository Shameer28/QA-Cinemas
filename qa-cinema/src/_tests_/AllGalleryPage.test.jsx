import AllGalleryPage from "../components/AllGalleryPage/AllGalleryPage";
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import PosterLink from "../components/sharedcomponents/PosterLink";

describe("All Gallery Page Testing", () => {

    it("Snapshot Check" , () => {
        const instance = create(<AllGalleryPage/>).toJSON();
        expect(instance).toMatchSnapshot();
    })

    it("Post Link are created from API", () => {
        render(<AllGalleryPage/>)
        const meh = screen.getAllByText("fast and furious 9");
        console.log(meh)
        expect(meh).toBe(3);
    })

})
