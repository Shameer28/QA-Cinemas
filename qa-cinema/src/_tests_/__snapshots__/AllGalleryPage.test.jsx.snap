// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Gallery Page Testing Snapshot Check 1`] = `
<div
  className="background"
>
  <div
    className="container"
  >
    <br />
    <h2
      style={
        Object {
          "color": "white",
          "fontWeight": "400",
          "fontfamily": "isonormregular, sans-serif",
          "letterSpacing": ".15em",
          "lineHeight": "1.1",
          "textTransform": "uppercase",
        }
      }
    >
      Now Showing
    </h2>
    <br />
    <div>
      <div
        className="row"
      >
        <p>
          Loading
        </p>
      </div>
    </div>
    <br />
  </div>
</div>
`;
